translators/DirectionalLight.cpp

```diff
@@ -0,0 +1,310 @@
+
+     //Update prim attribute values to maya node attibute values
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setVec3(mayaObj,m_pointWorld,pointWorld[0], pointWorld[1], pointWorld[2]), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setAngle(mayaObj,m_lightAngle,MAngle(angle,MAngle::Unit::kRadians)), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setVec3(mayaObj,m_color, color[0], color[1], color[2]), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_intensity, intensity), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_exposure, exposure), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_diffuse, diffuse), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_specular, specular), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setBool(mayaObj,m_normalize, normalize), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setBool(mayaObj,m_enableColorTemperature, enableColorTemperature), errorString);
+
+    return status;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+bool DirectionalLight::updateUsdPrim(const UsdStageRefPtr& stage, const SdfPath& usdPath, const MObject& mayaObj)
+{
+    const char* const errorString = "DirectionalLightTranslator: error getting maya directional light parameters";
+    float pointWorld[3] = {1.0,1.0,1.0};
+    MAngle angle;
+    float intensity = 0.0;
+    float exposure = 0.0;
+    float diffuse = 1.0;
+    float specular = 1.0;
+    bool normalize = false;
+    float color[3] = {1.0,1.0,1.0};
+    bool enableColorTemperature = false;
+
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getVec3(mayaObj, m_pointWorld, pointWorld), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getAngle(mayaObj, m_lightAngle, angle), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getVec3(mayaObj, m_color, color), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getFloat(mayaObj, m_intensity, intensity), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getFloat(mayaObj, m_exposure, exposure), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getFloat(mayaObj, m_diffuse, diffuse), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getFloat(mayaObj, m_specular, specular), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getBool(mayaObj, m_normalize, normalize), errorString);
+    AL_MAYA_CHECK_ERROR2(AL::usdmaya::utils::DgNodeHelper::getBool(mayaObj, m_enableColorTemperature, enableColorTemperature), errorString);
+
+    bool result;
+    UsdLuxDistantLight usdLight = UsdLuxDistantLight::Define(stage, usdPath);
+
+    if(!(pointWorld[0] == 1.0 && pointWorld[1]== 1.0 && pointWorld[2]==1.0))
```

Comment:
If you mix and match floats and doubles, C++ dictates that the data will always be converted to double to do the comparison. Here's a simple example: https://godbolt.org/z/FLTZjB

```suggestion
    if(!(pointWorld[0] == 1.0f && pointWorld[1] == 1.0f && pointWorld[2] == 1.0f))
```

Reply:
thanks for the example, I will change it.